op-chain-ops/cmd/check-delta/check-delta.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# Check if a directory path is provided
+# Directory must contain the output of batch_decoder's reassemble command
+if [ -z "$1" ]; then
+    echo "Usage: $0 /path/to/directory"
+    exit 1
+fi
+
+directory_path=$1
+valid_count=0
+invalid_count=0
+invalid_channels=()
+
+# Loop over every .json file in the specified directory
+for file in "$directory_path"/*.json; do
+    # If channel is ready, all batches must valid.
+    # If delta is activated, all batch types must be span batch.
```

Comment:
I'm not sure that this condition actually makes sense for checking delta's activation. It's possible to keep using old batches post delta.

Reply:
@trianglesphere You are right. Delta hardfork does not prohibit op-node to decode singular batches(old batches). The comment must be revised. My initial thought was, if we activate op-batcher with span batch enabled, every batch written in L1 will be in the form of span batch.

We must consider the scenario when op-batcher is span batch enabled but for some reason it could not write span batches to L1. The script should at least warn that it could not find span batches at all at L1.

My proposal is the following:
- Check all ready channels' batches are valid(`InvalidBatches = false`).
- Batch types can be singular batches or span batches. Count number of occurrences for each batch type.
- If number of observed span batch is zero, warn.
   - Example warn message: `Could not find valid span batches from L1. Cannot confirm delta is activated`.

Reply:
yea, simply reporting the number of different type of valid batches on L1 would be helpful. I agree that we should error on seeing invalid batches. Then it is up to the user to see that valid span batches are there

Reply:
> yea, simply reporting the number of different type of valid batches on L1 would be helpful. I agree that we should error on seeing invalid batches. Then it is up to the user to see that valid span batches are there

@trianglesphere I wrote the generalized version and named as `aggregate_batches.sh`. Also updated the initial comment(https://github.com/ethereum-optimism/optimism/pull/8417#issue-2023588514).

However, after when I finished writing every logic in bash, I feel like I should rewrite in go, or other better language. How do you think? Maybe followup PR will do this(add multiprocessing etc) and avoid error prone bash script.